File: week05_240708/서진경/BOJ_240714_2108.java
package silver;

import java.io.*;
import java.util.*;

public class boj_통계학_2108 {
	static BufferedReader br;
	static StringBuilder sb = new StringBuilder();
	public static void main(String[] args) throws IOException {
		br = new BufferedReader(new InputStreamReader(System.in));
		int N = Integer.parseInt(br.readLine());
		int[] arr = new int[N];
		int[] mode = new int[8001];
		int sum = 0;
		for(int i=0; i<N; i++) {
			arr[i] = Integer.parseInt(br.readLine());
			mode[arr[i]+4000]++;
			sum+=arr[i];
		}
		Arrays.sort(arr);
		
		sb.append((int) Math.round((double)sum/N)).append("\n");
		sb.append(arr[N/2]).append("\n");
		sb.append(findMode(mode)).append("\n");
		sb.append(arr[N-1]-arr[0]);
		
		System.out.println(sb);
		
	}
	
	public static int findMode(int[] mode) {
		int max = Integer.MIN_VALUE;
		int res = 0;
		boolean flag = false;
		for (int i = 0; i <8001; i++) {
		    if (mode[i] > max) {
		    	res = i - 4000;
		        max = mode[i];
		        flag = true;
		    } else if (mode[i] == max && flag == true) {
		        res = i - 4000;
		        flag = false;
		    }
		}
		return res;
	}
}

File: 2024/week05_240708/박준영/BOJ_240714_2108.py
import sys

I = sys.stdin.readline

N = int(I())

arr = []

for i in range(N):
    arr.append(int(I()))

arr.sort()

# 산술평균
print(round(sum(arr)/N))

# 중앙값
print(arr[N//2])

# 최빈값
dictionary = dict()
for i in arr:
    if i in dictionary:
        dictionary[i] += 1
    else:
        dictionary[i] = 1
max_value = max(dictionary.values())
max_arr = []

for j in dictionary:
    if max_value == dictionary[j]:
        max_arr.append(j)

if len(max_arr) > 1:
    print(max_arr[1])
else:
    print(max_arr[0])

# 범위
print(max(arr)-min(arr))